utils/src/main/kotlin/DeclaredLicenseProcessor.kt

```diff
@@ -105,7 +105,7 @@ object DeclaredLicenseProcessor {
 
         declaredLicenses.forEach { declaredLicense ->
             process(declaredLicense, declaredLicenseMapping)?.let {
-                processedLicenses[declaredLicense] = it
+                processedLicenses[declaredLicense] = it.disjunctiveNormalForm()
```

Comment:
I don't think we should use the DNF here, it can be very verbose for complex expressions (although they almost never appear in declared licenses in real life). For example `(a OR b) AND (c OR d)` would become `(a AND c) OR (a AND d) OR (b AND c) OR (b AND d)`. If we want that `MIT AND Apache-2.0` is shown sorted as `Apache-2.0 AND MIT` we could introduce a  separate sorting function that does not change the structure of the expression.

Reply:
Ok, I could probably drop that bit. But how about the code snippet above, which always sorts the licenses in the DNF?

Reply:
I don't see a reason not to do this.

Reply:
Well, despite that feedback coming a bit late, like the commit message says "This helps to compare different SPDX expressions". IMO a "normal form" should also be normalized in this regard, so that it is possible to compare two semantically equal DNFs also by their string representations.